File: programs/first-unique-character-in-a-string.js
var firstUniqChar = function(s) {
  var times = {}, result = -1, i;
  for (i = 0; i < s.length; ++i) {
    times[s[i]] = times[s[i]] ? ++times[s[i]] : 1;
  }

  for (i = 0; i < s.length; ++i) {
    if (1 === times[s[i]]) {
      result = i;
      break;
    }
  }

  return result;
};

console.log(firstUniqChar('leetcode'));
console.log(firstUniqChar('loveleetcode'));

File: drafts/first-unique-character-in-a-string.go
package main

import "fmt"

func firstUniqChar(s string) int {
	var times = make(map[string]int)
	result := -1

	for _, v := range s {
		times[string(v)]++
	}

	for i, v := range s {
		if times[string(v)] == 1 {
			result = i
			break
		}
	}

	return result
}

func main() {
	fmt.Println(firstUniqChar("leetcode"))
	fmt.Println(firstUniqChar("loveleetcode"))
}
